src/rgw/rgw_sal.cc

```diff
@@ -150,8 +150,14 @@ rgw::sal::Store* StoreManager::init_storage_provider(const DoutPrefixProvider* d
     const auto& data_path =
       g_conf().get_val<std::string>("rgw_sfs_data_path");
     ldpp_dout(dpp, 0) << "sfs init!" << dendl;
-    rgw::sal::SFStore *store =
-      new rgw::sal::SFStore(cct, data_path);
+    rgw::sal::SFStore *store = nullptr;
+    try {
+      store = new rgw::sal::SFStore(cct, data_path);
+    } catch (std::exception & e) {
+      ldpp_dout(dpp, 0) << e.what()
+                        << dendl;
+      return nullptr;
```

Comment:
does it make sense to return `nullptr` here?
is the caller protected; in case of nullptr what happens, does the process exit gracefully?

Reply:
This is the code calling `StoreManager::init_storage_provider`

```c+++
    store = StoreManager::get_storage(this, g_ceph_context,
					 rgw_store,
					 run_gc,
					 run_lc,
					 run_quota,
					 run_sync,
					 g_conf().get_val<bool>("rgw_dynamic_resharding"));

    if (!store) {
      mutex.lock();
      init_timer.cancel_all_events();
      init_timer.shutdown();
      mutex.unlock();

      derr << "Couldn't init storage provider (RADOS)" << dendl;
      return -EIO;
    }
```
When I tested this I got that `Couldn't init storage provider (RADOS)` message and `rgw` just exits.